client/files.js

```diff
@@ -50,15 +53,34 @@ filedrop.Files.prototype.list = function() {
 
 
 /**
- * Upload a file.
+ * Upload a file. Uses raw xhr rather than $http to access progress events.
```

Comment:
nit: instead of describing implementation, let's describe that the returned promise sends progress updates (and what the type of those updates is).